include/seqan3/core/type_list/traits.hpp

```diff
@@ -21,6 +22,19 @@
 namespace seqan3::pack_traits::detail
 {
 
+/*!\brief Implementation for seqan3::pack_traits::find.
+ * \tparam query_t  The type you are searching for.
+ * \tparam pack_t   The type pack.
+ * \returns The position of the first occurence of `query_t` in `pack_t` or `-1` if it is not contained.
+ * \ingroup type_list
+ */
+template <typename query_t, typename ...pack_t>
+constexpr ptrdiff_t find()
+{
+    ptrdiff_t i = 0;
+    return ((SEQAN3_IS_SAME(query_t, pack_t) ? false : ++i) && ...) ? -1 : i;
```

Comment:
I couldn't find anything whether or not the compiler sees this already and replaces it. So I guess we have to stick with this 😞